sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/Range.java

```diff
@@ -19,20 +19,32 @@
 
     private T minValue;
     private T maxValue;
+    private boolean minInclusive;
+    private boolean maxInclusive;
 
     public Range() {
         super();
     }
 
     public Range(String jsonString) {
         super(jsonString);
+        this.minValue = (T) super.get(Range.MIN_PROPERTY);
+        this.maxValue = (T) super.get(Range.MAX_PROPERTY);
+        this.minInclusive = super.getBoolean(Range.IS_MIN_INCLUSIVE_PROPERTY);
+        this.maxInclusive = super.getBoolean(Range.IS_MAX_INCLUSIVE_PROPERTY);
     }
 
     public Range(T min, T max, boolean isMinInclusive, boolean isMaxInclusive) {
-        this.setMin(min);
-        this.setMax(max);
-        this.setMinInclusive(isMinInclusive);
-        this.setMaxInclusive(isMaxInclusive);
+        this.minValue = min;
```

Comment:
Earlier we were setting these values using setMin(min), which will set the property values in property bag. In this PR, we are not setting the property bag values, but just the local values ?

Reply:
correct. they are only need to be in propertyBag when toJson is invoked() which is what we are doing now

Reply:
sounds good, thanks.